lib/auth/auth.go

```diff
@@ -4756,10 +4756,20 @@ func (a *Server) isMFARequired(ctx context.Context, checker services.AccessCheck
 			return nil, trace.Wrap(notFoundErr)
 		}
 
+		autoCreate, databaseRoles, err := checker.CheckDatabaseRoles(db)
```

Comment:
similar idea to the other comment, for a database that doesn't have admin user set, or a database that is not even postgres, as long this database matches this auto-create role, database-user matcher will be skipped. It's ok if this is designed/intended behaviour. Though users have to be very careful when crafting these auth-create roles.